screensharing/render.js

```diff
@@ -31,66 +31,109 @@ class ScreenShareRenderHook {
         this._iframe.addEventListener('load', this._onIframeApiLoad);
     }
 
+    // this is a rewrite of the sendRequest method on the transport
```

Comment:
Ugh, so we don't currently respond to requests we don't know about? That's bad...

Reply:
I think this is too much. I know it's more brittle, but doesn't matter since there is a fallback. Let's use a timeout please. Given this price is only paid once, it shouldn't be a huge deal...

Reply:
Also, 10 secodns is waaay to much, we should get an answer within 2 secods I think.

Reply:
A timeout alone would not help, we have to abort the promise

Reply:
This is why I wrote this function so that I can pass the controller signal in the promise constructor